src/lib/widget/_utilities/api.utilities.ts

```diff
@@ -0,0 +1,253 @@
+import {
+  Config,
+  FRUser,
+  SessionManager,
+  TokenManager,
+  UserManager,
+  type GetTokensOptions,
+} from '@forgerock/javascript-sdk';
+import HttpClient from '@forgerock/javascript-sdk/lib/http-client';
+import { get } from 'svelte/store';
+import type { z } from 'zod';
+
+// Import the stores for initialization
+import configure from '$lib/sdk.config';
+
+import type { HttpClientRequestOptions } from '@forgerock/javascript-sdk/lib/http-client';
+
+// Import store types
+import type { JourneyOptions, Modal, Response } from '../interfaces';
+import type { JourneyStore, JourneyStoreValue } from '$journey/journey.interfaces';
+import type { OAuthStore, OAuthTokenStoreValue } from '$lib/oauth/oauth.store';
+import type { partialConfigSchema } from '$lib/sdk.config';
+import type { UserStore } from '$lib/user/user.store';
+
+export function widgetApiFactory(modal?: Modal) {
+  let journeyStore: JourneyStore;
+  let oauthStore: OAuthStore;
+  let userStore: UserStore;
+
+  let returnError: (response: Response) => void;
+  let returnResponse: (response: Response) => void;
+
+  const configuration = {
+    set(options: z.infer<typeof partialConfigSchema>): void {
+      // Set base config to SDK
+      // TODO: Move to a shared utility
+      configure({
+        // Set some basics by default
+        ...{
+          // TODO: Could this be a default OAuth client provided by Platform UI OOTB?
+          clientId: 'WebLoginWidgetClient',
+          // TODO: If a realmPath is not provided, should we call the realm endpoint and detect a likely default?
+          // https://backstage.forgerock.com/docs/am/7/setup-guide/sec-rest-realm-rest.html#rest-api-list-realm
+          realmPath: 'alpha',
+          // TODO: Once we move to SSR, this default should be more intelligent
```

Comment:
What's SSR?

Reply:
I know we addressed this in our meeting, but just to respond here: SSR is Server Side Rendering.